### PR TITLE
#1715 API to get Switch detail by switch id via NB API.

### PR DESCRIPTION
**Pull request by dpoltavets**
Closes: #1715.

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/1773)

### CODE REVIEW

File: services/src/messaging/src/main/java/org/openkilda/messaging/nbtopology/request/GetSwitchRequest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.openkilda.messaging.nbtopology.annotations.ReadRequest;
+import org.openkilda.model.SwitchId;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+
+@SuppressWarnings("squid:MaximumInheritanceDepth")
+@ReadRequest
+@Getter
+public class GetSwitchRequest extends SwitchesBaseRequest {
```
Comment (sergii-iakovenko) [inline services/src/messaging/src/main/java/org/openkilda/messaging/nbtopology/request/GetSwitchRequest.java]: This is a message payload and should implement toString.

File: services/src/northbound/src/main/java/org/openkilda/northbound/service/impl/SwitchServiceImpl.java [hunk 2]
```diff
@@ -112,6 +113,23 @@
                         .collect(Collectors.toList()));
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CompletableFuture<SwitchDto> getSwitch(SwitchId switchId) {
+        final String correlationId = RequestCorrelationId.getId();
```
Comment (sergii-iakovenko) [inline services/src/northbound/src/main/java/org/openkilda/northbound/service/impl/SwitchServiceImpl.java]: In-line please as used only once.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/PersistenceOperationsBolt.java [hunk 3]
```diff
@@ -66,6 +67,11 @@ protected void handleInput(Tuple input) throws AbstractException {
                     e.getMessage(),
                     "ISL does not exist.");
             getOutput().emit(StreamType.ERROR.toString(), input, new Values(data, correlationId));
+        } catch (SwitchNotFoundException e) {
```
Comment (sergii-iakovenko) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/PersistenceOperationsBolt.java]: Something wrong with this. Why generic PersistenceOperationsBolt should care about specific SwitchNotFoundException?

**Comment by dpoltavets** [comment]
___
*[services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/PersistenceOperationsBolt.java, line 70 at r1](https://reviewable.io/reviews/telstra/open-kilda/1773#-LUCPaybfZJOHh4mCs2R-r1-70:-LUFLrnCD8gkNhbcBOK7:b-jux2j0) ([raw file](https://github.com/telstra/open-kilda/blob/eae52bc6041b2c310effc9d43c5ee31a35764207/services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/PersistenceOperationsBolt.java#L70)):*
<details><summary><i>Previously, timofei-durakov (Timofey Durakov) wrote…</i></summary><blockquote>

I believe it's done here this way to be aligned with IslNotFoundException. But these 2 exceptions are 2 specific to be used in base class. 
</blockquote></details>

If we catch this exception earlier, we will send 2 messages to NB (ErrorMessage and List <InfoData>). Or we will need to send ErrorMessage and throw an exception again in order not to send the list.

**Comment by dpoltavets** [comment]
___
*[services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/PersistenceOperationsBolt.java, line 70 at r1](https://reviewable.io/reviews/telstra/open-kilda/1773#-LUCPaybfZJOHh4mCs2R-r1-70:-LUnoD5BA1gnFP8AqTlH:b-gqb940) ([raw file](https://github.com/telstra/open-kilda/blob/eae52bc6041b2c310effc9d43c5ee31a35764207/services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/PersistenceOperationsBolt.java#L70)):*
<details><summary><i>Previously, sergii-iakovenko (Sergii Iakovenko) wrote…</i></summary><blockquote>

Can we catch a specific exception and wrap it into MessageException? So the bolt will care about MessageException only?
</blockquote></details>

Yes, we can. Fixed.

**Comment by dpoltavets** [comment]
___
*[services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/PersistenceOperationsBolt.java, line 69 at r4](https://reviewable.io/reviews/telstra/open-kilda/1773#-LUoo9eABr6MFGA2zpqP:-LUslXmU3mrCXV4OZHkL:bt2vr1b) ([raw file](https://github.com/telstra/open-kilda/blob/5d7071f124ace4ca06c92cc16946af44085b4ef9/services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/PersistenceOperationsBolt.java#L69)):*
<details><summary><i>Previously, sergii-iakovenko (Sergii Iakovenko) wrote…</i></summary><blockquote>

The error type should be taken from the exception. I think wrong MessageException is used here. Please, check CrudBolt.handle* methods
</blockquote></details>

Fixed.

**Pull request merged**